stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/application/App.kt

```diff
@@ -1,35 +1,24 @@
 package io.getstream.chat.sample.application
 
 import android.app.Application
-import com.getstream.sdk.chat.Chat
-import io.getstream.chat.android.client.notifications.handler.NotificationConfig
 import io.getstream.chat.sample.BuildConfig
-import io.getstream.chat.sample.R
 import io.getstream.chat.sample.data.dataModule
-import loginModule
+import io.getstream.chat.sample.feature.login.loginModule
+import io.getstream.chat.sample.feature.users.usersModule
 import org.koin.android.ext.android.inject
 import org.koin.android.ext.koin.androidContext
 import org.koin.android.ext.koin.androidLogger
 import org.koin.core.context.startKoin
 
 class App : Application() {
     private val appConfig: AppConfig by inject()
+    private val chatInitializer: ChatInitializer by inject()
 
     override fun onCreate() {
         super.onCreate()
         DebugMetricsHelper().init()
         initKoin()
-        Chat.Builder(appConfig.apiKey, this).apply {
-            offlineEnabled = true
-            val notificationConfig =
-                NotificationConfig(
-                    firebaseMessageIdKey = "message_id",
-                    firebaseChannelIdKey = "channel_id",
-                    firebaseChannelTypeKey = "channel_type",
-                    smallIcon = R.drawable.ic_chat_bubble
-                )
-            notificationHandler = SampleNotificationHandler(this@App, notificationConfig)
-        }.build()
+        chatInitializer.init(appConfig.apiKey)
```

Comment:
I'm just thinking if it won't be misleading that we initialize the Chat twice (second time in LoginViewModel). WDYT?

Reply:
Good point, maybe let's remove it from `App.kt`

Reply:
Yes, just a temp solution. OfflineSyncFirebaseMessagingService crashes the app on start because we don't have apiKey set. Will fix that in another PR.

Reply:
One thing to keep in mind here is that having these multiple sign-in options complicates the code of our SDK integration. For someone new to the SDK, the extra `ChatInitializer` we create, and initializing it in multiple places might be hard to understand at first.

For most clients, we just want them to initialize everything in their `Application` subclass, once, don't we? With these changes, our sample will no longer show off that approach.

I don't have a solution, because we do need the multiple sign-in options here, I'm just pointing out the potential issue of the sample no longer being that good of a sample for clients.

Reply:
I don't see the point in using this custom login screen. For developer considering the sample app it is easier to modify AppConfig with their api key and user token.

Reply:
I updated the PR and removed chat initialization from App. To prevent the app from crashing had to add a null check into `OfflineSyncFirebaseMessagingService.onNewToken()`.

Reply:
Agree, I don't have a better idea of how to handle that but I think we should leave Chat initialization in App class and give an explanation in `LoginViewModel` why do we need to initialize it for the second time.

Reply:
I'm leaning towards what @adasiewiczr proposed above ^, as it shows the regular init we expect people to do. Samples are important 😄.

Reply:
I've added again Chat init to the App class and added some explanation to the viewmodels